api/kubevirt.io/resource/pci/nas/v1alpha1/client/client.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	nascrd "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/nas/v1alpha1"
+	nasclient "kubevirt.io/kubevirt-dra-driver/pkg/kubevirt.io/resource/clientset/versioned/typed/nas/v1alpha1"
+)
+
+type Client struct {
+	nas    *nascrd.NodeAllocationState
+	client nasclient.NasV1alpha1Interface
+}
+
+func New(nas *nascrd.NodeAllocationState, client nasclient.NasV1alpha1Interface) *Client {
+	return &Client{
+		nas,
+		client,
+	}
+}
+
+func (c *Client) GetOrCreate(ctx context.Context) error {
+	err := c.Get(ctx)
+	if err == nil {
+		return nil
+	}
+	if errors.IsNotFound(err) {
+		return c.Create(ctx)
+	}
+	return err
+}
+
+func (c *Client) Create(ctx context.Context) error {
+	crd := c.nas.DeepCopy()
```

Comment:
Is this needed? You are overwriting in the next line

Reply:
The `DeepCopy` is necessary to create a separate copy of the object that can be modified and sent to the API without affecting the original object (c.nas).

The` Create(ctx, crd, metav1.CreateOptions{}) `call sends this copy to the Kubernetes API, which will modify the object